fedimint-api/src/db/mod.rs

```diff
@@ -247,6 +261,77 @@ impl Drop for CommitTracker {
     }
 }
 
+/// ModuleDatabaseTransaction is a wrapper around IsolatedDatabaseTransaction that
+/// consumes an existing DatabaseTransaction. This allows entire Databases to be
+/// isolated and calling begin_transaction will always produce a ModuleDatabaseTransaction,
+/// which is isolated from other modules by prepending a prefix to each key.
+struct ModuleDatabaseTransaction<'a> {
+    dbtx: DatabaseTransaction<'a>,
```

Comment:
Yeah I initially tried this, but ran into issue with the borrow checker when returning the new DatabaseTransaction (since IsolatedDatabaseTransaction borrows an exisitng DatabaseTransaction).

This is the only workable solution I've been able to come up with but if anyone comes up with a different solution we can certainly refactor.